closed/src/java.base/share/classes/com/sun/crypto/provider/NativeCipherBlockChaining.java

```diff
@@ -260,6 +260,21 @@ int encrypt(byte[] plain, int plainOffset, int plainLen,
             throw new ProviderException("Internal error in input buffering");
         }
 
+        /**
+         * OpenSSL doesn't support overlapping buffers.
+         * If plaintext and ciphertext byte arrays are overlapping, make a copy of plain
+         */
+        byte[] copyOfInput = null;
+        if (plain == cipher) {
+            if ((plainOffset >= cipherOffset) ||
```

Comment:
Seems this part should also check the length to ensure there is an overlap.

Reply:
the returned ciphertext's length is undetermined and may span all remaining space starting from cipherOffset.

Reply:
Pls add a comment to explain this.

Reply:
this java code invokes the OpenSSL routine EVP_CipherUpdate which is a generic function that can handle multiple modes of encryption/decryption. Tracking the output buffer size and ensuring it is sufficient is encryption mode dependent and is up to the code that uses OpenSSL. In our case, the code in [CipherCore.java ](https://github.com/ibmruntimes/openj9-openjdk-jdk/blob/42e7b85c8a915406a79fdef29d7180e5a06f16b9/src/java.base/share/classes/com/sun/crypto/provider/CipherCore.java#L578) manages the needed sizes and may buffer input if the output buffer is not sufficient. Turns out for AES-CBC mode update operation, the input/output buffers sizes should be identical. I will modify the check to reflect that.